src/main/java/org/cloudfoundry/promregator/cfaccessor/ReactiveCFAccessorImpl.java

```diff
@@ -392,4 +409,114 @@ private void setupPaginatedRequestFetcher() {
+
+		return this.paginatedRequestFetcher.performGenericRetrieval(RequestType.SPACE, key, spacesRequest, sr ->
+																		this.cloudFoundryClient.spacesV3().list(sr),
+																	this.requestTimeoutSpace);
+	}
+
+	@Override
+	public Mono<org.cloudfoundry.client.v3.spaces.ListSpacesResponse> retrieveSpaceIdsInOrgV3(String orgId) {
+		PaginatedRequestGeneratorFunctionV3<org.cloudfoundry.client.v3.spaces.ListSpacesRequest> requestGenerator = (resultsPerPage, pageNumber) ->
+			org.cloudfoundry.client.v3.spaces.ListSpacesRequest.builder()
+							 .organizationId(orgId)
+							 .perPage(resultsPerPage)
+							 .page(pageNumber)
+							 .build();
+
+		PaginatedResponseGeneratorFunctionV3<org.cloudfoundry.client.v3.spaces.SpaceResource, org.cloudfoundry.client.v3.spaces.ListSpacesResponse> responseGenerator = (list, numberOfPages) ->
+			org.cloudfoundry.client.v3.spaces.ListSpacesResponse.builder()
+							  .addAllResources(list)
+							  .pagination(Pagination.builder().totalPages(numberOfPages).totalResults(list.size()).build())
+							  .build();
+
+
+		return this.paginatedRequestFetcher.performGenericPagedRetrievalV3(RequestType.SPACE_IN_ORG, orgId, requestGenerator,
+																		 r -> this.cloudFoundryClient.spacesV3().list(r),  this.requestTimeoutSpace, responseGenerator);
+	}
+
+	@Override
+	public Mono<org.cloudfoundry.client.v3.applications.ListApplicationsResponse> retrieveAllApplicationIdsInSpaceV3(String orgId, String spaceId) {
+		String key = String.format("%s|%s", orgId, spaceId);
+
+		PaginatedRequestGeneratorFunctionV3<org.cloudfoundry.client.v3.applications.ListApplicationsRequest> requestGenerator = (resultsPerPage, pageNumber) ->
+			org.cloudfoundry.client.v3.applications.ListApplicationsRequest.builder()
+								   .organizationId(orgId)
+								   .spaceId(spaceId)
+								   .perPage(resultsPerPage)
+								   .page(pageNumber)
+								   .build();
+
+		PaginatedResponseGeneratorFunctionV3<org.cloudfoundry.client.v3.applications.ApplicationResource, org.cloudfoundry.client.v3.applications.ListApplicationsResponse> responseGenerator = (list, numberOfPages) ->
+			org.cloudfoundry.client.v3.applications.ListApplicationsResponse.builder()
+									.addAllResources(list)
+									.pagination(Pagination.builder().totalPages(numberOfPages).totalResults(list.size()).build())
+									.build();
+
+		return this.paginatedRequestFetcher.performGenericPagedRetrievalV3(RequestType.ALL_APPS_IN_SPACE, key, requestGenerator,
+																		 r -> this.cloudFoundryClient.applicationsV3().list(r), this.requestTimeoutAppInSpace, responseGenerator);
+	}
+
+	@Override
+	public Mono<GetSpaceResponse> retrieveSpaceV3(String spaceId) {
+		// This API has drastically changed in v3 and does not support the same resources. This call for a space summary will probably
```

Comment:
This API will need to be reworked if we intend to do full V3 support. See `retrieveRoutesForAppId` as it is a new API call that is not implemented yet.

Reply:
This is fine for now